chainer/links/normalization/batch_normalization.py

```diff
@@ -204,9 +223,12 @@ def __init__(self, size=None, decay=0.9, eps=2e-5, dtype=None,
             self._initialize_params(size)
 
     def _initialize_params(self, shape):
-        self.avg_mean = numpy.zeros(shape, dtype=self._dtype)
+        dtype = self._dtype
+        self.avg_mean = _init_array(self._initial_avg_mean, 0, shape, dtype)
+        del self._initial_avg_mean
```

Comment:
How about assigning `None`?
I think it's not a good practice to add/remove this kind of attributes in ad-hoc manner.